### DISCUSSION TITLE
Why do I not receive more than "email" from the traits?

### DISCUSSION THREAD

**Discussion by AdamNording** [ory/kratos#3588] [Q&A]
Hello, I'm setting up a React app with Kratos and I have a lot of trubbel getting for instance given_name, last_name and picture from the traits. I only get the **email** when I console.log the ory session. I have given all the permissions in AzureAD to be able to get info about the logged in user. I can log in and use 2FA and I can see in Azure that I successfully logged in. 

Is there something wrong with my setup or is there something very specific in Azure AD that needs to be configured to make it work with Kratos? 

kratos.yml:

<img width="826" alt="image" src="https://github.com/ory/kratos/assets/49994496/aaa8a5d6-fe78-4a55-b140-c9245cf4e36c">

identity.schema.json

<img width="527" alt="image" src="https://github.com/ory/kratos/assets/49994496/c0a026d9-47a3-4aa6-87ad-48734322812d">

identity.jsonnet

<img width="683" alt="image" src="https://github.com/ory/kratos/assets/49994496/055f6df1-d944-4203-96e1-db176ee76b0c">

**Comment by vinckr** [comment]
Hello @AdamNording 

Hm did you create the user anew? The traits etc. get only mapped when you link the OIDC method.

Do all the traits get mapped correctly in the logged in user identity?

**Reply by AdamNording** [reply]
Hello @vinckr 

No it is my existing company user. Okey but what do you mean by "link the OIDC method"? I send the JSONnet file as a base64 url to the mapper, thought that that would be enough. Where do I link the OIDC method? 

No, I only see the the email field in the traits. The other fields aren't visible in the response. But if I for instance do like this in the identities.schema.json file and follow the same pattern in the JSONnet file: 


`"name": {`
`   "type": "object",`
`   "properties": {`
`        "first": {`
`             "type": "string",`
`             "title": "First name"`
`        },`
`        "last": {`
`             "type": "string",`
`             "title": "Last name"`
`        }`
`     }`
`}`


I can see the object "name" in the traits response, but the object is empty. Neither "first" or "last" is set.

**Reply by vinckr** [reply]
Sorry I was not clear, I mean link the OIDC when the user is signing up with Azure or linking it to an existing identity.

So when you create a new user through the Azure login the traits should be set. 

Do you get the claims from Azure, last_name etc?

**Reply by AdamNording** [reply]
Ah okey no worries. No I'm using an existing user, which have a profile picture, last_name and given_name. 

I get no claims other than the email.
